internal/cache/controller_obj_cache.go

```diff
@@ -461,15 +461,12 @@ func (c *AviObjCache) AviPopulateAllPkiPRofiles(client *clients.AviClient, pkiDa
 			utils.AviLog.Warnf("Incomplete pki data unmarshalled, %s", utils.Stringify(pki))
 			continue
 		}
-		checksum := lib.SSLKeyCertChecksum(*pki.Name, string(*pki.CaCerts[0].Certificate), "")
-		if lib.GetGRBACSupport() && pki.Markers != nil {
-			checksum += lib.ObjectLabelChecksum(pki.Markers)
-		}
+
 		pkiCacheObj := AviPkiProfileCache{
 			Name:             *pki.Name,
 			Uuid:             *pki.UUID,
 			Tenant:           lib.GetTenant(),
-			CloudConfigCksum: checksum,
+			CloudConfigCksum: lib.SSLKeyCertChecksum(*pki.Name, string(*pki.CaCerts[0].Certificate), "", pki.Markers),
```

Comment:
@chauhanshubham : Markers field introduced in 20.1.5. So for controller version 20.1.4 and less, I think this will cause exception.

Reply:
+1 to @akshayhavile . Unless it's nil because of us using the 20.1.5 SDK.

Reply:
I believe it would cause an Exception if pki is nil, which won't be the case here. We are using the 20.1.5 sdk which would probably omit the `Markers` key if it's not fetched in case of a 20.1.4 controller (or less). 
We were doing a `pki.Markers != nil` check prior to this, which has been moved inside the checksum function. So in case of an incompatible controller, we would just be passing `nil` here.

Reply:
verified with Avi v20.1.4